### PR TITLE
pagination and store Procedure Completed

### PR DESCRIPTION
**Pull request by Blawal-SE**

### CODE REVIEW

File: SchoolApplication/School.Data/SchoolContext.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            var password_param = new SqlParameter("@Password", password);
+            var confirmpassword_param = new SqlParameter("@ConfirmPassword", confirmpassword);
+            var email_param = new SqlParameter("@Email", email);
+            
+            return ((IObjectContextAdapter)this).ObjectContext.ExecuteStoreQuery<Nullable<Decimal>>("AddStudent  @Name ,@FNname ,@Email ,@Phone, @Dob ,@Password ,@ConfirmPassword", name_param, fname_param, email_param, phone_param, dob_param, password_param, confirmpassword_param).First();
+               
+        }
+        public virtual void AddCourses(int StudentId, List<int> courseslist)
+        {
+
```
Comment (hassan22khan) [inline SchoolApplication/School.Data/SchoolContext.cs:41]: unwanted spaces

File: SchoolApplication/School.Repository/StudentRepository.cs [hunk 2]
```diff
@@ -13,35 +13,23 @@ namespace School.Repository
...
     {
         #region student Repository
-        public List<StudentDtoPost> GetStudents()
+        public List<StudentDtoPost> GetStudents(Pager pager)
         {
-            List<StudentDtoPost> DtoList = new List<StudentDtoPost>();
+          //  List<StudentDtoPost> DtoList = new List<StudentDtoPost>();
             try
             {
+
```
Comment (hassan22khan) [inline SchoolApplication/School.Repository/StudentRepository.cs]: unwanted spaces

File: SchoolApplication/School.Repository/StudentRepository.cs [hunk 3]
```diff
@@ -13,35 +13,23 @@ namespace School.Repository
     public class StudentRepository : BaseRepository<Student>
     {
         #region student Repository
-        public List<StudentDtoPost> GetStudents()
+        public List<StudentDtoPost> GetStudents(Pager pager)
         {
-            List<StudentDtoPost> DtoList = new List<StudentDtoPost>();
+          //  List<StudentDtoPost> DtoList = new List<StudentDtoPost>();
```
Comment (hassan22khan) [inline SchoolApplication/School.Repository/StudentRepository.cs]: remove the comments or comment why you have still kept these comments

File: SchoolApplication/School.Repository/StudentRepository.cs [hunk 4]
```diff
@@ -83,23 +69,37 @@ public StudentDtoPost FindStudent(int id)
...
+                    s.Email = studentDto.Email;
+                    s.Password = studentDto.Password;
+                    s.ConfirmPassword = studentDto.ConfirmPassword;
+                    db.Students.Add(s);
+                    List<StudentCourse> courseList = new List<StudentCourse>();
+                    if (studentDto.Courses != null)
                     {
-                        stdCourse.CourseId = Convert.ToInt32(course);
-                        db.StudentCourses.Add(stdCourse);
+                        foreach (var Course in studentDto.Courses)
```
Comment (hassan22khan) [inline SchoolApplication/School.Repository/StudentRepository.cs:83]: variables need to be in camel Case

File: SchoolApplication/School.Repository/View/Pager.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+{
+    public class Pager
+    {
+        public int draw { get; set; }
+        public int start { get; set; }
+        public int length { get; set; }
+        public List<Columns> columns { get; set; }
+        public List<Order> order{ get; set; }
+        public Search search { get; set; }
+
```
Comment (hassan22khan) [inline SchoolApplication/School.Repository/View/Pager.cs:16]: unwanted spaces

File: SchoolApplication/School.Repository/UserRolesRepository.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                using (var context = new SchoolContext())
+                {
+                    var user = (from u in context.Users.Where(x => x.UserName == username && x.Password == password)
+                                select new UserRoleDto()
+                                {
+                                    Id = u.Id,
+                                    FullName = u.FullName,
+                                    UserName = u.UserName,
+                                    Password = u.Password,
+                                    ConfirmPassword = u.ConfirmPassword,
```
Comment (Nauman7) [inline SchoolApplication/School.Repository/UserRolesRepository.cs:27]: why saving confirm password into the db?

**Review by Nauman7 — APPROVED** [review]

**Pull request merged**